File: src/lib/software/inno_setup.py
from src.lib.software import BaseSoftware

class InnoSetup(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.download_name = 'innosetup.exe'
        self.silent_install_args = ['/VERYSILENT', '/SUPPRESSMSGBOXES', '/NORESTART', '/SP-']

    def resolve_download_url(self):
        self.url_resolved.emit('https://jrsoftware.org/download.php/is.exe?site=1')

File: packages/app/src/services/software/definitions/innoSetup.ts
import { ok, err } from 'neverthrow';
import { DownloadUrlResolveError, SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'inno-setup';
	public name = 'Inno Setup';
	public category = [SoftwareCategory.Development];
	public downloadName = 'innosetup.exe';
	public shouldCacheUrl = true;
	public icon = 'inno-setup.png';
	public homepage = 'https://jrsoftware.org/isinfo.php';

	public async resolveDownloadUrl() {
		const res = await fetch('https://jrsoftware.org/download.php/is.exe?site=1');
		if (!res.ok) {
			return err(DownloadUrlResolveError.HTTPResponseError);
		}

		return ok(res.url);
	}
}
